### PR TITLE
Google Gemini (Independant Publisher)

### PR DESCRIPTION
**Pull request by PriyaranjanKS**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 

#### POST `/{apiVersion}/models/{modelName}:generateContent`
This endpoint generates textual content based on input messages using the specified Google Gemini model.

#### Parameters
| Name            | Required | Description                                           |
|-----------------|----------|-------------------------------------------------------|
| `apiVersion`    | Yes      | The API version for the endpoint (e.g., v1beta).      |
| `modelName`     | Yes      | The name of the model used for generation (e.g., gemini-pro). |
| `contents`      | Yes      | The input messages for content generation.            |
| `safetySettings`| No       | Optional settings for content safety.                 |
| `generationConfig` | No   | Configuration for content generation, including randomness, token limits, and stop sequences. |

#### Request Body Structure
The request body should include the `contents` for generation, optional `safetySettings`, and `generationConfig`. The `generationConfig` can specify parameters like `maxOutputTokens`, `temperature`, `topP`, `topK`, and `stopSequences`.

#### Responses
- `200 OK`: Successful response with generated content.
- `422 Validation Error`: Indicates issues with request parameters or structure.

#### Usage

Generate Content 1

--

Generate Content 2

--
 

Generate Content 3

### CODE REVIEW

**Review by vmanoharas — COMMENTED** [review]
Hello @PriyaranjanKS, 

Can you please help us to understand why this is empty PR?

image

**Comment by PriyaranjanKS** [comment]
> Hello @PriyaranjanKS,
> 
> Can you please help us to understand why this is empty PR?
> 
> image

@vmanoharas As part of the initial stage of the connector publishing process, I have opened the PR for proposal approval of a non existing connector and have attached the implemented screenshots. By tomorrow will be adding the swagger files.

File: independent-publisher-connectors/Google Gemini/readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (vmanoharas) [inline independent-publisher-connectors/Google Gemini/readme.md:192]: Can you please update readme.md file as per the template below. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md

**Review by vmanoharas — COMMENTED** [review]
Hi @PriyaranjanKS, 

Please update readme.md file.

File: independent-publisher-connectors/Google Gemini/readme.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+## Publisher: Priyaranjan KS, Vidya Sagar Alti [Tata Consultancy Services]
+
+## Overview
+Custom connector for Google Gemini, providing advanced AI text generation functionalities.
+
+## Prerequisites
+An API key from Google is required to access the Google Gemini API.
+
+### Obtaining an API Key
```
Comment (vmanoharas) [inline independent-publisher-connectors/Google Gemini/readme.md]: Please change this to Obtaining Credentials

**Review by vmanoharas — COMMENTED** [review]
Hi @PriyaranjanKS, 

Please resolve one move review comment.

File: independent-publisher-connectors/Google Gemini/readme.md (hunk 2 above)
Comment (PriyaranjanKS) [reply independent-publisher-connectors/Google Gemini/readme.md]: @vmanoharas Sure, Will do that. Mean time, I am doing one more round of regression testing, will keep you posted in a day or two. Thanks.

File: independent-publisher-connectors/Google Gemini/readme.md (hunk 1 above)
Comment (PriyaranjanKS) [reply independent-publisher-connectors/Google Gemini/readme.md:192]: @vmanoharas This is completed.

File: independent-publisher-connectors/Google Gemini/readme.md (hunk 2 above)
Comment (PriyaranjanKS) [reply independent-publisher-connectors/Google Gemini/readme.md]: This is also done.

Comment (PriyaranjanKS) [reply independent-publisher-connectors/Google Gemini/readme.md]: @vmanoharas The swagger and readme updates are completed.

**Review by vmanoharas — APPROVED** [review]
Hello @PriyaranjanKS 
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by vmanoharas** [comment]
[[certify-connector]]

**Comment by vmanoharas** [comment]
Hello @PriyaranjanKS ,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Comment by vmanoharas** [comment]
Hi @PriyaranjanKS, 

One of the user of this connector getting below error.  Could you assit? 

image

Error Message: The connector 'Google Gemini (Independent Publisher)' returned an HTTP error with code 400. Inner Error: * GenerateContentRequest.contents: contents is not specified Error Code: ConnectorRequestFailure Conversation Id: 253e53f3-d52a-4557-b78c-2367a996cf71 Time (UTC): 2026-05-20T15:04:45.147Z
 
Any guidance you can provide on what it might be missing will be very helpful.

**Pull request merged**